### PR TITLE
Add edit button functionality and redesign for general feedback tab

### PR DESCRIPTION
**Pull request by katjanakosic**
- Added an edit button for the general feedback cells
- Adjusted the score design (incl. color and picker range)
- Adjusted overall general feedback cell design

### CODE REVIEW

File: Themis/Views/Assessment/CorrectionSidebar/EditFeedbackView.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        let feedback: Feedback
+
+        if feedbackID == nil {
+            feedback = Feedback(feedbackText: feedbackText, score: score)
+        } else {
+            feedback = Feedback(id: feedbackID, feedbackText: feedbackText, score: score)
+        }
+        feedbackModel.saveFeedback(feedback: feedback)
+        updateStates()
+    }
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/EditFeedbackView.swift:91]: These two functions should be moved to feedbackViewModel

File: Themis/ViewModels/CorrectionSidebar/FeedbackViewModel.swift [hunk 2]
```diff
@@ -37,10 +36,16 @@ public class FeedbackViewModel: ObservableObject {
         return feedback.score
     }
 
+    public func saveFeedback(feedback: Feedback) {
+        self.feedbacks.removeAll(where: { $0.id == feedback.id })
+        feedbacks.append(feedback)
```
Comment (AlexExcellence) [inline Themis/ViewModels/CorrectionSidebar/FeedbackViewModel.swift]: This results in the edited feedback being moved to the bottom of the list, which can be confusing.
This should work
```suggestion
        if let index = feedbacks.firstIndex(where: { $0.id == feedback.id }) {
             feedbacks[index] = feedback
        } else {
            feedbacks.append(feedback)
        }
```

File: Themis/Views/Assessment/CorrectionSidebar/EditFeedbackView.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                Text(title)
+                    .font(.largeTitle)
+                Spacer()
+                Button {
+                    withAnimation {
+                        save()
+                        showEditFeedback = false
+                    }
+                } label: {
+                    Text("Save")
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/EditFeedbackView.swift:40]: Indicate that the Button is disabled by setting a dynamic foregroundColor
```suggestion
                    Text("Save")
                    .foregroundColor(feedbackText.isEmpty ? .gray : .blue)
```

**Review by AlexExcellence — APPROVED** [review]
Looks fine, added some minor things we should change.

**Comment by botrufus** [comment]
As already mentioned in a comment, to allow conveniently submitting feedback to Artemis the feedback model you created had to be deleted as it didn't conform to the fields which were needed for the request. Additionally due to the restructuring of the view models, feedback view model was redundant as feedbacks can simply be put in to the assessment view model. So except for the design changes, everything functionality wise will unfortunately be discarded and/or translated to the new model. Let's talk about this today during the meeting and figure out how we can best do this. :)

**Pull request closed without merging**